internal/replacements/replacements.go

```diff
@@ -257,6 +265,51 @@ func resourcesNeedsReplacement(desired []corev1.Container, current []corev1.Cont
 	return desiredCPURequests.Cmp(*currentCPURequests) == 1 || desiredMemoryRequests.Cmp(*currentMemoryRequests) == 1
 }
 
+// fileSecurityContextChanged checks for changes in the effective security context by checking that there are no changes
+// to the following SecurityContext (or PodSecurityContext) fields:
+// RunAsGroup, RunAsUser, FSGroup, or FSGroupChangePolicy
+// See https://github.com/FoundationDB/fdb-kubernetes-operator/issues/208 for motivation
+// only makes sense if both pods have containers with matching names
+func fileSecurityContextChanged(desired, current *corev1.Pod) bool {
+	// first check for FSGroup or FSGroupChangePolicy changes as that cannot be overridden at container level
+	// (if pod security context is identical, skip these checks)
+	if (desired.Spec.SecurityContext != nil || current.Spec.SecurityContext != nil) &&
+		!equality.Semantic.DeepEqualWithNilDifferentFromEmpty(desired.Spec.SecurityContext, current.Spec.SecurityContext) {
+		if desired.Spec.SecurityContext == nil { // check if changed non-nil -> nil
+			if current.Spec.SecurityContext.FSGroup != nil || current.Spec.SecurityContext.FSGroupChangePolicy != nil {
+				return true
+			}
+		} else if current.Spec.SecurityContext == nil { // check if changed nil -> non-nil
+			if desired.Spec.SecurityContext.FSGroup != nil || desired.Spec.SecurityContext.FSGroupChangePolicy != nil {
+				return true
+			}
+		} else { // both pod security contexts are defined so check they are the same
+			if !equality.Semantic.DeepEqualWithNilDifferentFromEmpty(desired.Spec.SecurityContext.FSGroup, current.Spec.SecurityContext.FSGroup) ||
+				!equality.Semantic.DeepEqualWithNilDifferentFromEmpty(desired.Spec.SecurityContext.FSGroupChangePolicy, current.Spec.SecurityContext.FSGroupChangePolicy) {
+				return true
+			}
+		}
+	}
+	// check for RunAsUser and RunAsGroup changes (have to check with container settings, since that can override pod settings)
+	for _, desiredContainer := range desired.Spec.Containers {
+		for _, currentContainer := range current.Spec.Containers {
+			if desiredContainer.Name == currentContainer.Name {
+				desiredEffectiveSecCtx := securitycontext.DetermineEffectiveSecurityContext(desired, &desiredContainer)
+				currentEffectiveSecCtx := securitycontext.DetermineEffectiveSecurityContext(current, &currentContainer)
+				if equality.Semantic.DeepEqual(desiredEffectiveSecCtx, currentEffectiveSecCtx) {
+					continue
```

Comment:
```suggestion
					break
```

If we are at this code line, we can skip the further iteration of the inner loop as we already found the matching container.